### PR TITLE
#13256 Serve translated basemap style based on user's language

### PR DESCRIPTION
**Pull request by azapasnik**

### CODE REVIEW

File: src/test/java/io/kontur/disasterninja/dto/layer/LayerSummaryDtoTest.java [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+        assertEquals("https://d.n/style_ar.json", dto.getMapboxStyle());
+    }
+
+    @Test
+    public void testMapBoxStyle_AcceptLanguageFirstWin() {
+        Layer layer = createLayer();
+        layer.setMapboxStyles(JsonUtil.readJson("{\"url\":\"https://d.n/style_{lang}.json\",\"lang\":[\"en\",\"ar\"]}", ObjectNode.class));
+
+        MockHttpServletRequest request = new MockHttpServletRequest();
+        request.addHeader("accept-language", "ar,en;q=0.9,ru-BY;q=0.8,ru;q=0.7,en-GB;q=0.6,en-US;q=0.5");
```
Comment (Komzpa) [inline src/test/java/io/kontur/disasterninja/dto/layer/LayerSummaryDtoTest.java:51]: wait, does this mean that basemap will be translated if frontend language is not selected? seems wrong. we don't control accept-language and thus if front end forgets to set user-language then you will get heisenbugs that depend on user's computer's language.

I'd rather let us ERROR in the logs if no user-language provided. The logic of choosing app locale based on browser locale may need to go to frontend instead, once and overridable.

File: src/main/java/io/kontur/disasterninja/dto/layer/LayerSummaryDto.java [hunk 2]
```diff
@@ -22,12 +30,82 @@ public class LayerSummaryDto {
...
+
+        Enumeration<Locale> acceptLanguage = request.getLocales();
+        while (acceptLanguage.hasMoreElements()) {
+            String language = acceptLanguage.nextElement().getLanguage();
+            if (supportedLanguages.contains(language)) {
+                return url.replace("{lang}", language);
+            }
+        }
+
+        if (supportedLanguages.contains("en")) {
```
Comment (Komzpa) [inline src/main/java/io/kontur/disasterninja/dto/layer/LayerSummaryDto.java:76]: is there a logic for non-translated basemap, where just name= tag is used and labels are different for each country?

**Review by Komzpa — CHANGES_REQUESTED** [review]
too much fallback logic

File: src/main/java/io/kontur/disasterninja/dto/layer/LayerSummaryDto.java (hunk 2 above)
Comment (azapasnik) [reply src/main/java/io/kontur/disasterninja/dto/layer/LayerSummaryDto.java:76]: Nope. en is intended to be the last fallback style. And I didn't generate such style for basemap. Although there is some style_day.json that uses name for labels but it wasn't used for DN.

File: src/test/java/io/kontur/disasterninja/dto/layer/LayerSummaryDtoTest.java (hunk 1 above)
Comment (azapasnik) [reply src/test/java/io/kontur/disasterninja/dto/layer/LayerSummaryDtoTest.java:51]: > wait, does this mean that basemap will be translated if frontend language is not selected?

Wasn't it initial plan to support accept-language with the list of languages?

>  if front end forgets to set user-language 

You know what? We don't set any language by default for non logged users and for users who didn't set language explicitly in app settings. So basemap will be translated according to accept-language for these cases. You are right. Seems like not the best solution.

**Review by i1shilyaev — APPROVED** [review]

**Review by Komzpa — APPROVED** [review]

**Pull request merged**